async-nats/src/connector.rs

```diff
@@ -35,6 +35,7 @@ pub(crate) struct ConnectorOptions {
     pub(crate) tls_client_config: Option<rustls::ClientConfig>,
     pub(crate) auth: Authorization,
     pub(crate) no_echo: bool,
+    pub(crate) connection_timeout: Option<Duration>,
```

Comment:
i completely agree, my original implementation did not include this. and i only included it because i thought it would be more ergonomic for developers who are interested in the "old" behavior.

i'll remove it.

Reply:
i've merged/pushed the requested change

Reply:
> If someone really wants to wait for a long time, high timeout values can be set, but I believe there is no use case for no-timeout scenario.

Yup I agree.